biz.aQute.launcher/src/aQute/launcher/plugin/ProjectLauncherImpl.java

```diff
@@ -369,6 +370,12 @@ public Jar executable() throws Exception {
 		Manifest m = new Manifest();
 		Attributes main = m.getMainAttributes();
 
+		logger.debug("Use Embedded launcher");
+		// contains all whitelisted properties which should not be removed in any way
+		Set<String> whitelistProperties = new HashSet<>(Arrays.asList(MAIN_CLASS, EMBEDDED_RUNPATH));
```

Comment:
Since this is a static list value, why create a Set and why create it each invocation. 

```java
private static final List<String> whitelistProperties = Arrays.asList(MAIN_CLASS, EMBEDDED_RUNPATH);
```

should be fine as a static variable.

Reply:
You are right - I just wanted to have it in the same line as setting the manifest to ensure if a additional property is added changing the whitelist is not forgotten - but i changed it 🙂.